### ISSUE TITLE
Hash.sha3() result is different from sha3 other service

### ISSUE THREAD

**Issue by Twibap** [web3j/web3j#778]
hi, i have a trouble with sha3 result in web3j
Hash.sha3() result is different from [here](https://emn178.github.io/online-tools/keccak_256.html)
and also checksum result is different from [here](https://tokenmarket.net/ethereum-address-validator)

i'm trying to checksum address "0xB4760d454eAEA3FEe39EA0a65DE4c108a3960582"
![2018-11-09 12 49 20](https://user-images.githubusercontent.com/26022935/48211040-fe525c00-e3bb-11e8-96da-92f2d555bf85.png)

this is my android snippet
```
public void onClick(View view) {
        String address = inputAddress.getText().toString();
        String hash = Hash.sha3( address );
        String checksumAddress = AddressUtils.toChecksumAddress(address);

        viewAddress.setText(address);
        viewHash.setText( hash );
        viewChecksum.setText(checksumAddress);
}
```
```
class AddressUtils{
    static String toChecksumAddress(String address){
        address = address.toLowerCase();
        Log.d(TAG, "toChecksumAddress: source   -> "+address);
        String hash = Hash.sha3(address);
        Log.d(TAG, "toChecksumAddress: Hash     -> "+hash);

        String result = "0x";
        for(int i = 2 ; i < address.length() ; i++ ){   // i = 2; after prefix 

            if( Integer.parseInt( Character.toString(hash.charAt(i)),  16 ) >= 8){
                result += Character.toString(address.charAt(i)).toUpperCase() ;
            } else {
                result += Character.toString(address.charAt(i));
            }
        }

        Log.d(TAG, "toChecksumAddress: checksum -> "+ result);
        return result;
    }
}
```
```
dependencies {
    implementation 'org.web3j:utils:3.3.1-android'
}
```

Anyone can help me? Thank you!

**Comment by iikirilov** [comment]
`Hash.sha256()`...
Also the example address you are trying to checksum is already checksummed...

**Comment by Twibap** [comment]
> `Hash.sha256()`...
> Also the example address you are trying to checksum is already checksummed...

my point is `Hash.sha3`'s result is different from other services
and `Hash.sha256` is also different.
![2018-11-09 12 48 11](https://user-images.githubusercontent.com/26022935/48242278-d1369580-e41d-11e8-8a66-be7f7ceef18d.png)

as you can see checksum result is different
Checksum address from Metamask : 
`0xB4760d454eAEA3FEe39EA0a65DE4c108a3960582`
Checksum address from Hash.sha256 of web3j : 
`0xB4760d454Eaea3fee39EA0A65De4C108A3960582`
Checksum address from Hash.sha3 of web3j : 
`0xb4760d454EaeA3FeE39EA0a65dE4c108A3960582`

the example is lowercase in `toChecksumAddress()` function
and will be use to valid address. like this
```
    static boolean isValidChecksumAddress(String address){
        return toChecksumAddress(address).equals(address);
    }
```

**Comment by iikirilov** [comment]
related #486 #476 #311 #208 #194

**Comment by Twibap** [comment]
Solved by #194.
if you want to address checksum, avoid using `hexStringToByteArray()`
use this
```
String address = "0xB4760d454eAEA3FEe39EA0a65DE4c108a3960582";
address = address.toLowerCase().replace("0x", "");
byte[] hash = Hash.sha3( address.getBytes( StandardCharsets.UTF-8 ) );
```
